lib/spotlight/dor/indexer.rb

```diff
@@ -53,6 +54,29 @@ def resource druid
 
     private
 
+    # add the box number to solr_doc as box_ssi field (note: single valued!)
+    #   data in location/physicalLocation or in relatedItem/location/physicalLocation
+    # TODO:  push this up to stanford-mods gem?  or should it be hierarchical series/box/folder?
+    def add_box(sdb, solr_doc)
+      # see spec for data from actual collections
+      #   _location.physicalLocation should find top level and relatedItem
+      box_num = sdb.smods_rec._location.physicalLocation.map do |node|
+        val = node.text
+        # note that this will also find Flatbox or Flat-box
+        match_data = val.match(/Box ?:? ?([^,|]+)/i)
```

Comment:
I think we can simplify this logic down to a single regular expression, maybe something like:

```ruby
/Box ?:? ?([^,|(Folder)]+)/i
```